java/main/src/main/java/com/saucelabs/saucebindings/CITools.java

```diff
@@ -64,10 +50,23 @@ public static String getBuildName() {
    * @return the variable number of the current build
    */
   public static String getBuildNumber() {
-    if (!getCiToolName().equalsIgnoreCase("unknown")) {
-      String buildNumberKey = BUILD_VALUES.get(getCiToolName()).get(1);
-      return SystemManager.get(buildNumberKey);
+    return getCiTool().getBuildNumber();
+  }
+
+  private static CITool getCiTool() {
+    if (ciTool != null) {
+      return ciTool;
+    } else if (SystemManager.get("SAUCE_BUILD_NAME") != null) {
```

Comment:
Why do you only check for `SAUCE_BUILD_NAME` and not the other two?

Reply:
You can get away with the defaults for the other 2, but if Build Name isn't set, there's no reason to track tool name or build number

Reply:
No, this is a good point. I updated all the classes to check for defaults for everything.